image.c

```diff
@@ -293,14 +300,223 @@ bool img_load_gif(img_t *img, const fileinfo_t *file)
+	if (fread((uint8_t*)data.bytes, 1, file_size, webp_file) != file_size) {
+		error(0, 0, "%s: Error opening webp image", file->name);
+		free((uint8_t*)data.bytes);
+		return false;
+	}
+
+	// Setup the WebP Animation Decoder
+	if (!WebPAnimDecoderOptionsInit(&opts)) {
+		// Version mismatch in WebP library
+		error(0, 0, "%s: Error opening webp image", file->name);
+		free((uint8_t*)data.bytes);
+		return false;
+	}
+	opts.color_mode = MODE_BGRA;
+	// This could cause problems on some systems and will require more testing
+	// than I can do (multithreaded decoding)
+	opts.use_threads = true;
+	if ((dec = WebPAnimDecoderNew(&data, &opts)) == NULL) {
+		// Parsing error, invalid operation, or memory error
+		error(0, 0, "%s: Error opening webp image", file->name);
+		free((uint8_t*)data.bytes);
+		return false;
+	}
+	if (!WebPAnimDecoderGetInfo(dec, &info)) {
+		error(0, 0, "%s: Error opening webp image", file->name);
+		WebPAnimDecoderDelete(dec);
+		free((uint8_t*)data.bytes);
+		return false;
+	}
+	demux = WebPAnimDecoderGetDemuxer(dec);
+
+	// Get global information for the image
+	flags = WebPDemuxGetI(demux, WEBP_FF_FORMAT_FLAGS);
+	img->w = WebPDemuxGetI(demux, WEBP_FF_CANVAS_WIDTH);
+	img->h = WebPDemuxGetI(demux, WEBP_FF_CANVAS_HEIGHT);
+	img->alpha = ALPHA_LAYER;
+	img->multi.cap = info.frame_count;
+	img->multi.sel = 0;
+	img->multi.frames = emalloc(info.frame_count * sizeof(img_frame_t));
+
+	// Load (and decode) the frames of the image (also works with images that
+	// have only 1 frame)
+	img->multi.cnt = 0;
+	while (WebPAnimDecoderGetNext(dec, &buf, &ts)) {
+		im = imlib_create_image_using_copied_data(
+				info.canvas_width, info.canvas_height, (DATA32*)buf);
+		imlib_context_set_image(im);
+		imlib_image_set_format("webp");
+		// Get an iterator of this frame - used for frame info (duration, etc.)
+		WebPDemuxGetFrame(demux, img->multi.cnt+1, &iter);
+		imlib_image_set_has_alpha((flags & ALPHA_FLAG) > 0);
+		// Store info for this frame
+		img->multi.frames[img->multi.cnt].im = im;
+		delay = iter.duration > 0 ? iter.duration : DEF_WEBP_DELAY;
+		img->multi.frames[img->multi.cnt].delay = delay;
+		img->multi.length += img->multi.frames[img->multi.cnt].delay;
+		img->multi.cnt++;
+	}
+	WebPAnimDecoderDelete(dec);
+	WebPDemuxReleaseIterator(&iter);
+
+	if (img->multi.cnt > 1) {
+		imlib_context_set_image(img->im);
+		imlib_free_image();
+		img->im = img->multi.frames[0].im;
+	} else if (img->multi.cnt == 1) {
+		imlib_context_set_image(img->multi.frames[0].im);
+		imlib_free_image();
+		img->multi.cnt = 0;
+	}
+
+	imlib_context_set_image(img->im);
+	imlib_image_set_format("webp");
+
+	free((uint8_t*)data.bytes);
+	return !err;
+}
+
+Imlib_Image load_webp_firstframe_im(const fileinfo_t *file)
+{
+	struct stat st;
+
+	if (!file->name || !file->path)
+		return NULL;
+	if (!STREQ(strrchr(file->path, '.'), ".webp"))
+		return NULL;
+	if (access(file->path, R_OK) != 0 || stat(file->path, &st) != 0 || !S_ISREG(st.st_mode))
+		return NULL;
```

Comment:
Once again, `check_add_file` already checks for this. Also why should we disallow loading a webp file if the extension isn't `.webp`.

It's perfectly fine to load an image named `img` without any extension.

Reply:
after some tests the patch has a general problem with images without extensions, trying to load any image without extension, not just webp (static or animated) will cause a segfault.

Reply:
after further testing this very check is the culprit of sxiv segfaulting on every extensionless image, without it a extensionless animated webp will load and play just fine without any problem, however the stdout will be littered by a `sxiv: /path/to/non_webp.img: Error opening webp image` message on every non webp image